flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -276,6 +277,16 @@ public long getLatencyTrackingInterval() {
         return configuration.get(MetricOptions.LATENCY_INTERVAL).toMillis();
     }
 
+    @PublicEvolving
+    public Optional<MemorySize> getGlobalAggregationBufferSize() {
```

Comment:
You are mixing layers here. Global aggregation is a concept of the SQL layer and thus should only change classes in the `flink-table` module. Configuration doesn't need to be added to `ExecutionConfig`. You can pass parameters to the runtime class during planning in the corresponding ExecNode.